lib/services/nativescript-migration-service.ts

```diff
@@ -85,92 +115,190 @@ export class NativeScriptMigrationService implements IFrameworkMigrationService
 				return;
 			}
 
-			let projectDir = this.$project.getProjectDir().wait();
-			let tnsModulesProjectPath = path.join(projectDir, this.$projectConstants.NATIVESCRIPT_APP_DIR_NAME, "tns_modules");
-			let appResourcesRequiredPath = path.join(projectDir, this.$projectConstants.NATIVESCRIPT_APP_DIR_NAME, this.$staticConfig.APP_RESOURCES_DIR_NAME);
-			let appResourcesObsoletePath = path.join(projectDir, this.$staticConfig.APP_RESOURCES_DIR_NAME);
-			let backupName = `${tnsModulesProjectPath}.backup`;
-			let shouldRollBackAppResources = false;
+			this.throwIfVersionDeleted(currentFrameworkVersion).wait();
 
-			// Check if current version is supported one. We cannot migrate ObsoleteVersions
-			if(!_.contains(this.getSupportedVersions().wait(), currentFrameworkVersion)) {
-				this.checkIsVersionObsolete(currentFrameworkVersion).wait();
+			// Check if current version is supported one. We cannot migrate DeletedVersions
+			if (this.isDeleted(currentFrameworkVersion).wait()) {
```

Comment:
`throwIfVersionDeleted` will throw before this line, why do you need this check?